api/test/e2enew/upstream/upstream_test.go

```diff
@@ -0,0 +1,510 @@
+				 }],
+				 "type": "roundrobin"
+			 }`,
+			Headers:      map[string]string{"Authorization": base.GetToken()},
+			ExpectStatus: http.StatusOK,
+		}),
+		table.Entry("hit the route using upstream 1", base.HttpTestCase{
+			Object:       base.APISIXExpect(),
+			Method:       http.MethodGet,
+			Path:         "/hello",
+			ExpectStatus: http.StatusOK,
+			ExpectBody:   "hello world",
+			Sleep:        base.SleepTime,
+		}),
+		table.Entry("delete not exist upstream", base.HttpTestCase{
+			Object:       base.ManagerApiExpect(),
+			Method:       http.MethodDelete,
+			Path:         "/apisix/admin/upstreams/not-exist",
+			Headers:      map[string]string{"Authorization": base.GetToken()},
+			ExpectStatus: http.StatusNotFound,
+		}),
+		table.Entry("delete route", base.HttpTestCase{
+			Object:       base.ManagerApiExpect(),
+			Method:       http.MethodDelete,
+			Path:         "/apisix/admin/routes/1",
+			Headers:      map[string]string{"Authorization": base.GetToken()},
+			ExpectStatus: http.StatusOK,
+		}),
+		table.Entry("delete upstream", base.HttpTestCase{
+			Object:       base.ManagerApiExpect(),
+			Method:       http.MethodDelete,
+			Path:         "/apisix/admin/upstreams/1",
+			Headers:      map[string]string{"Authorization": base.GetToken()},
+			ExpectStatus: http.StatusOK,
+		}),
+		table.Entry("hit the route just deleted", base.HttpTestCase{
+			Object:       base.APISIXExpect(),
+			Method:       http.MethodGet,
+			Path:         "/hello",
+			ExpectStatus: http.StatusNotFound,
+			ExpectBody:   "{\"error_msg\":\"404 Route Not Found\"}\n",
+			Sleep:        base.SleepTime,
+		}),
+	)
+})
+
+var _ = ginkgo.Describe("Upstream chash remote addr", func() {
+	ginkgo.It("create chash upstream with key (remote_addr)", func() {
+		base.RunTestCase(base.HttpTestCase{
+			Object: base.ManagerApiExpect(),
+			Method: http.MethodPut,
+			Path:   "/apisix/admin/upstreams/1",
+			Body: `{
+				 "nodes": [{
+					 "host": "172.16.238.20",
+					 "port": 1980,
+					 "weight": 1
+				 },
+				 {
+					 "host": "172.16.238.20",
+					 "port": 1981,
+					 "weight": 1
+				 },
+				 {
+					 "host": "172.16.238.20",
+					 "port": 1982,
+					 "weight": 1
+				 }],
+				 "type": "chash",
+				 "hash_on":"header",
+				 "key": "remote_addr"
+			 }`,
+			Headers:      map[string]string{"Authorization": base.GetToken()},
+			ExpectStatus: http.StatusOK,
+		})
+	})
+
+	ginkgo.It("create route using the upstream just created", func() {
+		base.RunTestCase(base.HttpTestCase{
+			Object: base.ManagerApiExpect(),
+			Method: http.MethodPut,
+			Path:   "/apisix/admin/routes/1",
+			Body: `{
+				 "uri": "/server_port",
+				 "upstream_id": "1"
+			 }`,
+			Headers:      map[string]string{"Authorization": base.GetToken()},
+			ExpectStatus: http.StatusOK,
+			Sleep:        base.SleepTime,
+		})
+	})
+
+	ginkgo.It("hit routes(upstream weight 1)", func() {
+		t := ginkgo.GinkgoT()
+		//hit routes
+		basepath := "http://127.0.0.1:9080/"
```

Comment:
please use [`base.APISIXHost `](https://github.com/apache/apisix-dashboard/blob/master/api/test/e2enew/base/base.go#L40) instead of `http://127.0.0.1:9080`

Reply:
done.